### PR TITLE
Active storage

### PR DESCRIPTION
**Pull request by anjali-kr**
- Devise integration, Cloudinary & Active Storage setup done
- Model validations added

### CODE REVIEW

File: app/models/booking.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :offer
+
+  # address must be provided
+  validates :address, length: { minimum: 6 }
+  validates :booking_date, presence: true
+  # start time must be before end time
+  validates :start_time, :end_time, presence: true
+  validates :comment, presence: true
```
Comment (al3xandra33) [inline app/models/booking.rb]: I would leave the comment optional. (As it is also in AirBnB...) Maybe someone does not want to add a comment to the booking.

File: app/models/offer.rb [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
   has_many :bookings
+  has_one_attached :photo
+
+  CATEGORY = ['German', 'Italian', 'Vietnamese', 'French', 'Indian', 'Fusion'].freeze
+
+  # An offer must have a unique title and a description
+  validates :title, uniqueness: true, presence: true
+  validates :description, presence: true
+
+  validates :price, numericality: { only_integer: true }, presence: true
```
Comment (al3xandra33) [inline app/models/offer.rb]: the price is float. Not sure what only_integer does though..

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
+          <%= f.button :submit, "Log in", class: 'btn btn-primary' %>
+        </div>
+      <% end %>
 
-<%= render "devise/shared/links" %>
+      <%= render "devise/shared/links" %>
+
+    </div>
+  </div>
+</div>
```
Comment (al3xandra33) [inline app/views/devise/sessions/new.html.erb:27]: These are very small conflicts (indentation and the design of a button) but I guess we need to discuss before deleting one of them.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -15,8 +15,8 @@
   <body>
     <%= render 'shared/navbar' %>
 
-    <p class="notice"><%= notice %></p>
-    <p class="alert"><%= alert %></p>
```
Comment (al3xandra33) [inline app/views/layouts/application.html.erb:19]: I know we need to add these 2 lines from the slides.

File: app/views/pages/home.html.erb [hunk 5]
```diff
@@ -1,2 +1,3 @@
-<h1>Pages#home</h1>
```
Comment (al3xandra33) [inline app/views/pages/home.html.erb:1]: We should delete these 2 rows (the standard ones provided by Le Wagon)

File: app/views/pages/home.html.erb [hunk 6]
```diff
@@ -1,2 +1,3 @@
...
-<p>Find me in app/views/pages/home.html.erb</p>
```
Comment (al3xandra33) [inline app/views/pages/home.html.erb:2]: same here

File: app/views/shared/_flashes.html.erb [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+    <%= notice %>
+  </div>
+<% end %>
+<% if alert %>
+  <div class="alert alert-warning alert-dismissible fade show m-1" role="alert">
+    <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+    <%= alert %>
+  </div>
+<% end %>
```
Comment (al3xandra33) [inline app/views/shared/_flashes.html.erb:12]: what does this do? :D

File: app/views/pages/home.html.erb (hunk 6 above)
Comment (anjali-kr) [reply app/views/pages/home.html.erb:2]: these have been already deleted. Hence the - sign and red color

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (anjali-kr) [reply app/views/layouts/application.html.erb:19]: this is removed and replaced by 'shared/flashes'

File: app/views/devise/sessions/new.html.erb (hunk 3 above)
Comment (anjali-kr) [reply app/views/devise/sessions/new.html.erb:27]: This is only indentation, the form has been centered using bootstrap grid.

File: app/models/offer.rb (hunk 2 above)
Comment (anjali-kr) [reply app/models/offer.rb]: You're right. I need to check this again

File: app/views/shared/_flashes.html.erb (hunk 7 above)
Comment (anjali-kr) [reply app/views/shared/_flashes.html.erb:12]: Just displays the 'you have been logged in..' type messages in the bottom right corner. This was also taken from the lecture :)

File: app/models/booking.rb (hunk 1 above)
Comment (anjali-kr) [reply app/models/booking.rb]: Removed presence check for comment

**Review by al3xandra33 — APPROVED** [review]

**Pull request merged**